build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/EmbeddedProviderExtension.java

```diff
@@ -36,35 +35,25 @@ void impl(String implName, Project implProject) {
         String projectName = implProject.getName();
         String capitalName = capitalize(projectName);
 
-        String configName = "embedded" + capitalName;
-        Configuration implConfig = project.getConfigurations().create(configName);
+        Configuration implConfig = project.getConfigurations().detachedConfiguration(project.getDependencies().create(implProject));
         implConfig.attributes(attrs -> {
             attrs.attribute(ARTIFACT_TYPE_ATTRIBUTE, DIRECTORY_TYPE);
-            attrs.attribute(IMPL_ATTR, true);
+            attrs.attribute(EmbeddedProviderPlugin.IMPL_ATTR, true);
         });
 
-        var deps = project.getDependencies();
-        deps.registerTransform(UnzipTransform.class,
-            transformSpec -> {
-                transformSpec.getFrom().attribute(ARTIFACT_TYPE_ATTRIBUTE, JAR_TYPE).attribute(IMPL_ATTR, true);
-                transformSpec.getTo().attribute(ARTIFACT_TYPE_ATTRIBUTE, DIRECTORY_TYPE).attribute(IMPL_ATTR, true);
-                transformSpec.parameters(parameters -> parameters.getIncludeArtifactName().set(true));
-            });
-        deps.add(configName, implProject);
-
         String manifestTaskName = "generate" + capitalName + "ProviderManifest";
-        Directory generatedResourcesDir = project.getLayout().getBuildDirectory().dir("generated-resources").get();
+        Provider<Directory> generatedResourcesDir = project.getLayout().getBuildDirectory().dir("generated-resources");
         var generateProviderManifest = project.getTasks().register(manifestTaskName, GenerateProviderManifest.class);
         generateProviderManifest.configure(t -> {
-            t.getManifestFile().set(generatedResourcesDir.file("LISTING.TXT"));
+            t.getManifestFile().set(generatedResourcesDir.get().file("LISTING.TXT"));
```

Comment:
This can just be `generatedResourcesDir.file("LISTING.TXT")`; The `set()` method accepts a `Provider`.